Move a few vendor libraries to src/native/external

am11 opened the pull request:
Moved:
* brotli
* rapidjson
* zlib
* zlib-intel

am11 committed:
Move brotli to src/native/external

am11 committed:
Move rapidjson to src/native/external

am11 committed:
Move zlib to src/native/external

am11 commented:
cc @jkotas
* rapidjson was missing the version file, so added `rapidjson-version.txt`
* zlib-intel is not moved, only the one which we use on win-arm/arm64 in libs and on multiple targets in mono

The format is:
```sh
src/native/external/{libname}
src/native/external/{libname}-version.txt
src/native/external/{libname}.cmake # exception being rapidjson which is header-only lib with no build configs
```

@janvorli, I was thinking about moving libunwind here as well and refactoring our configurations in a `libunwind cmake` file without overwriting anything in vendor's tree. (of course after https://github.com/dotnet/runtime/pull/62092 is merged) :slightly_smiling_face:

am11 committed:
Move zlib-intel to src/native/external

am11 commented:
Red diffs are because one (out of three) copies of zlib are deleted from the repo.
cc @carlossanlop

jkotas commented on src/native/corehost/hostcommon/files.cmake:
```diff
@@ -7,7 +7,7 @@ include_directories(${CMAKE_CURRENT_LIST_DIR}/../json)
 # CMake does not recommend using globbing since it messes with the freshness checks
 list(APPEND SOURCES
-    ${CMAKE_CURRENT_LIST_DIR}/../json_parser.cpp
+    ${CMAKE_CURRENT_LIST_DIR}/../json/json_parser.cpp
```
Was there a technical reason for moving this file into a subdirectory? I think it was fine where it was.

am11 replied:
I've moved it back.
json/ directory had only one subdirectory which is now moved, so it felt that it's the right place for json_parser to be or (otherwise there was no reason for json in corehost/json/rapidjson).

jkotas reviewed (approved):
Looks great. Thank you!

am11 committed:
Move json_parser back to corehost root

am11 committed:
Cleanup trailing whitespace in changeset files
Except for the ones in vendor directory:
```sh
# git remote add dotnet https://github.com/dotnet/runtime && git pull --rebase dotnet main

if uname 2>/dev/null | grep -q Darwin; then
    space=" "
fi

git show --name-only --pretty="" HEAD...dotnet/main \
    ':!*external/brotli/*' ':!*external/rapidjson/*' ':!*external/zlib/*' ':!*external/zlib-intel/*' |
        xargs -I{} sh -c "test -f {} && sed -i$space'' 's/[[:space:]]*$//' {}"
```
